bundle/edu.gemini.itc.shared/src/main/java/edu/gemini/itc/shared/ConfigCreator.java

```diff
@@ -0,0 +1,201 @@
+package edu.gemini.itc.shared;
+
+import edu.gemini.spModel.config2.Config;
+import edu.gemini.spModel.config2.DefaultConfig;
+import edu.gemini.spModel.config2.ItemKey;
+import edu.gemini.spModel.gemini.gmos.GmosCommonType;
+import edu.gemini.spModel.guide.GuideProbe;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Using the ITC parameters creates a mini-config containing the items
+ * necessary for the observation overhead calculations.
+ * The config has number of steps equal to number of exposures. Even though
+ * just two steps would be enough to calculate overheads, I went for more
+ * straightforward solution, simulating the OT sequence, which I believe
+ * would make any future adds-ons easier to implement (and in case the ITC
+ * may become more sophisticated in future).
+ *
+ * Created by osmirnov on 11/28/17.
+ */
+
+public class ConfigCreator {
+    public static final ItemKey ObserveTypeKey = new ItemKey("observe:observeType");
+    public static final ItemKey ObserveClassKey = new ItemKey("observe:class");
+    public static final ItemKey ExposureTimeKey = new ItemKey("observe:exposureTime");
+    public static final ItemKey CoaddsKey = new ItemKey("observe:coadds");
+    public static final ItemKey InstInstrumentKey = new ItemKey("instrument:instrument");
+    public static final ItemKey ReadModeKey = new ItemKey("instrument:readMode");
+    public static final ItemKey AmpReadModeKey = new ItemKey("instrument:ampReadMode");
+    public static final ItemKey SlitWidthKey = new ItemKey("instrument:slitWidth");
+    public static final ItemKey DisperserKey = new ItemKey("instrument:disperser");
+    public static final ItemKey BuiltinROIKey = new ItemKey("instrument:builtinROI");
+    public static final ItemKey FPUKey = new ItemKey("instrument:fpu");
+    public static final ItemKey CcdXBinning = new ItemKey("instrument:ccdXBinning");
+    public static final ItemKey CcdYBinning = new ItemKey("instrument:ccdYBinning");
+    public static final ItemKey AmpGain = new ItemKey("instrument:gainChoice");
+    public static final ItemKey AmpCount = new ItemKey("instrument:ampCount");
+    public static final ItemKey DetectorManufacturerKey = new ItemKey("instrument:detectorManufacturer");
+    public static final ItemKey GuideWithPWFS2Key = new ItemKey("telescope:guideWithPWFS2");
+    public static final ItemKey TelescopeQKey = new ItemKey("telescope:q");
+    public static final ItemKey TelescopePKey = new ItemKey("telescope:p");
+    public static final ItemKey AOGuideStarTypeKey = new ItemKey("adaptive optics:guideStarType");
+    public static final ItemKey AOSystemKey = new ItemKey("adaptive optics:aoSystem");
+    public static final ItemKey AOFieldLensKey = new ItemKey("adaptive optics:fieldLens");
```

Comment:
I grepped through the codebase and these are defined multiple times in multiple files, which makes me a bit uncomfortable. For example, lots of these are already in `jsky.app.it.editor.seq.Keys`.

I wonder if we should be reusing these instead of redefining them repeatedly?

Reply:
I really wanted to reuse the ItemKeys, since as you noticed they are indeed defined multiple times throughout the ocs. However there are only few packages which are accessible from the ITC packages, and the OT packages are not among them. I remember introducing some ”circular dependencies”, probably while trying to add more dependencies to the ITC. So I went for redefining, as I couldn’t come up with a better solution. If you could suggest a way to do it otherwise, I can change it.